### ISSUE TITLE
System crash when creating runes index on Ubuntu

### ISSUE THREAD

**Issue by so7ow** [ordinals/ord#3852]
I experienced a repeated system crash when attempting to create a runes index with `ord 0.19.0`. All recent versions of ord have indexed without issue. I thought it might just be a failing disk or something (see crash details below), but now at least one other user on the Ordicord has reported a similar situation, so I thought I'd submit this issue. After each crash, I'd delete the partial REDB file and start again, and each time it would crash again. When I finally let the system repair the partial index and continue, I was able to complete the runes index. I'm happy to try to reproduce the crash and gather whatever logs would be helpful in diagnosing, if that's appropriate. Or maybe it is just a failing disk over here. 🤷🏻‍♂️

**My command:**
```
$ ord --index-runes --index index-runes-19.redb server
```

**My system:**
```
 $ free -m
               total        used        free      shared  buff/cache   available
Mem:           64095       29194         562           1       34338       34177
Swap:          65535        4029       61506

 $ cat /proc/cpuinfo          
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 158
model name      : Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
stepping        : 10
microcode       : 0xf6
cpu MHz         : 4578.741
cache size      : 12288 KB
physical id     : 0
siblings        : 12
core id         : 0
cpu cores       : 6
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 22

4 TB Samsung 990 PRO:
 $ df -h
Filesystem                         Size  Used Avail Use% Mounted on
tmpfs                              6.3G  1.7M  6.3G   1% /run
/dev/mapper/ubuntu--vg-ubuntu--lv  3.6T  1.8T  1.7T  53% /
tmpfs                               32G     0   32G   0% /dev/shm
tmpfs                              5.0M     0  5.0M   0% /run/lock
/dev/nvme0n1p2                     2.0G  255M  1.6G  14% /boot
/dev/nvme0n1p1                     1.1G   34M  1.1G   4% /boot/efi
tmpfs                              6.3G  4.0K  6.3G   1% /run/user/1000

 $ cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
```
**The crash:**
The system goes completely unresponsive and this was on the screen:
![4DA0647B-56C0-420C-964B-3208743AD7DA_1_201_a](https://github.com/user-attachments/assets/d93bf25c-70b4-4dc1-b96d-0d5a78bd55cb)

**Comment by Nutter79** [comment]
I'm experiencing the same issue.
Have attempted indexing with 0.19.0, each attempt after deleting index.redb file. Each time it crashes after a while.

![image](https://github.com/user-attachments/assets/8cc4fcf8-6aca-4a05-90c7-5f65ddd29cb0)

**Comment by Nutter79** [comment]
Just wanted to add that i've installed Windows 11 onto the same PC and have had no issues so far indexing.
Would have expected a crash after a few hours, but have been indexing now for a few days without any issues.

So using the same hardware, i'm experiencing issues when on Ubuntu but not on Windows.

**Comment by so7ow** [comment]
This issue still exists in 0.19.1 for me. Just like in 0.19.0, sats/runes/address index completes fine, runes/addresses index crashes somewhere after block 805000.


This command succeeds:

```ord --index-runes --index-sats --index-addresses --index index-runes-sats-addresses-19.redb server```

This command eventually crashes the whole system:

```ord --index-runes --index-addresses --index index-runes-addresses-19.redb server```

**Comment by so7ow** [comment]
This is resolved for me in 0.20.0
